tests/console/docker_compose.pm

```diff
@@ -0,0 +1,45 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test docker-compose installation
+#    Cover the following aspects of docker-compose:
+#      * package can be installed
+# Maintainer: Panagiotis Georgiadis <pgeorgiadis@suse.com>
+
+
+use base "consoletest";
+use testapi;
+use utils;
+use strict;
+
+# Setup the required testing environment
+sub setup {
+    # install the docker package if it's not already installed
+    zypper_call("in docker") if script_run("zypper se -x --provides -i docker | grep docker");
```

Comment:
just call the "in docker" unconditionally as zypper_call will make sure it's installed. If it's already installed it will skip over it. I bet this is even faster then first searching if the package that provides an application if not already installed